engines/sherlock/surface.h

```diff
@@ -0,0 +1,62 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef SHERLOCK_GRAPHICS_H
+#define SHERLOCK_GRAPHICS_H
+
+#include "common/rect.h"
+#include "graphics/surface.h"
+#include "sherlock/resources.h"
+
+namespace Sherlock {
+
+class Surface : public Graphics::Surface {
```

Comment:
Using public inheritance here seems a bit wacky. This allows code working on this Surface class to use `Graphics::Surface::create` which does not handle the `_freePixels` flag. This is potentionally an easy way to shoot yourself into the knee. It also allows you to do things like "Graphics::Surface foo = someSherlockSurface" which might behave oddly because of all this (can also be the case for passing parameters!).

I think it definitely be worthwhile to evaluate this decision again.